tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameDefaultValueService.java

```diff
@@ -50,23 +51,32 @@ protected String compute()
         String retVal = "";
 
         final String projectName = op().getProjectName().content( true );
-        
-        if ( projectName == null )
+
+        if( projectName == null )
         {
             return retVal;
         }
-        
-        final String projectTemplate = op().getProjectTemplateName().content( true );
-        
-        final String pName = projectName.replaceAll( "-", "" );
-        
-        final String finalProjectName = pName.replaceAll( "\\.", "" );
-        
-        final StringBuffer componentNameBuffer = new StringBuffer(finalProjectName);
-        
-        componentNameBuffer.append( projectTemplate );
-        
-        return componentNameBuffer.toString();
+
+        String projectTemplate = op().getProjectTemplateName().content( true );
+
+        if( projectTemplate != null )
+        {
+            final char[] tokens = new char[] { '-', '.', '_' };
+
+            String finalProjectName = WordUtils.capitalizeFully( projectName, tokens );
+
+            for( char token : tokens )
+            {
+                finalProjectName = finalProjectName.replaceAll( "\\" + token, "" );
+            }
+
+            final StringBuffer componentNameBuffer = new StringBuffer( finalProjectName );
+
+            componentNameBuffer.append( projectTemplate );
```

Comment:
We don't need to append the project template.  That was the way we did things in  plugins sdk, for now we should let the user's name be exactly what is used.